stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/Sprintf.java

```diff
@@ -95,19 +95,32 @@ public final void execute(final Context context) {
                 }
                 try {
                     char formatSpecifier = format.charAt(j);
+                    BRefType ref = args.getRefValue(k);
                     switch (formatSpecifier) {
                         case 'b':
                         case 'B':
                         case 'd':
                         case 'f':
-                            result.append(String.format("%" + padding + formatSpecifier, args.getRefValue(k).value()));
+                            if (ref == null) {
+                                throw BLangExceptionHelper.getRuntimeException(RuntimeErrors.ILLEGAL_FORMAT_CONVERSION,
+                                        format.charAt(j) + " != ()");
+                            } else {
```

Comment:
Let's remove the else part, since we are throwing the exception in the if condition.